### ISSUE TITLE
Automatically allowed cross-site cookies!?

### ISSUE THREAD

**Issue by privacyguy123** [arkenfox/user.js#1448]
🟥 https://github.com/arkenfox/user.js/wiki/5.2-Troubleshooting
- [x] I have read the troubleshooting guide, done the checks and confirmed this is caused by arkenfox
   - _unchecked issues ~~may~~ will be closed as invalid_

🟪 REQUIRED INFO
  - Browser version & OS: 100
  - Steps to Reproduce (STR): Browse Facebook then another website through a link on Facebook
  - Expected result: Not automatically allowed?
  - Actual result: Automatically allowed - no prompt
  - Console errors and warnings:
  - Anything else you deem worth mentioning:

---

![image](https://user-images.githubusercontent.com/50337995/168684083-ea1a52e9-53b0-4a9c-88fc-d42a0f64d3ed.png)

**Comment by Thorin-Oakenpants** [comment]
not seeing it - give me actual links

I can't see anything on FB because I don;t have an account, but I can load facebook.com and click all items in the footer and get a bunch of fb cookies. And I took a link from there to instagram. Not seeing anything - provide LINKs

**Comment by Thorin-Oakenpants** [comment]
Are you allowing a site exception for facebook in clear cookies + site data?

**Comment by privacyguy123** [comment]
Any link on a FB new feed: https://l.facebook.com/l.php?u=https%3A%2F%2Fwww.ultimate-guitar.com%2Fnews%2Fgeneral_music_news%2Fgwar_singer_calls_iron_maiden_a_bunch_of_british_old_women_explains_why_steve_harris_bothers_him.html%3Ffbclid%3DIwAR2IZshDEkOk8_7jdRPSyMvYnwclDJMoEj6F-Jvn9CgDaVqqSn_MzyvHz4s&h=AT0KorToAv7ls8MZXgWaMymoj3PSRyT8FeWMiNS4reGSeTImnej-O_usSucMIXJ_O3L_rzk3-ZqCqBlfk32VqAnjObxo6tGpArWy_1FrbJvhs9QJPVHu2OgMLu1f2vhvSQ&__tn__=H-R&c[0]=AT1Kf7HxMy5Igj4cwUMXqYUsPDCnPmcDmLw2Eg1ns9fgIuHV1PVREowMw5UG7662HqF4nTC9jMlKDO5V1TXyy_1ykF_vsp-VDU_T7sg7WEd6SEoEpOMx3HDqii_4-Z-2f3NuNdT8YoCabyNpn9iXjTRzic5OvLD6c_gIud4ASlMxBMvzHgkSymONpSCrxylNZV0Vm3XVDSkiW3nN

**Comment by privacyguy123** [comment]
> Are you allowing a site exception for facebook in clear cookies + site data?

Yes to save log in - is there a way to do this without allowing automatic cross site tracking?

**Comment by Thorin-Oakenpants** [comment]
I'll @wisniewskit just FYI, no need to do anything unless you want to confirm that `l.facebook.com` tracker being allowed is because of a sanitizing exception - I suspect it is because it's listed in the panel as an exception

i.e a case of [1767271](https://bugzilla.mozilla.org/show_bug.cgi?id=1767271)

---

I bounced around FB a lot more, got to see a heap of groups and local and services and live feeds and videos and all sorts (nightly with no extensions, sanitized on previous close, no site exceptions for anything, not cookies, not ETP, nada) and got 5 fb cookies (see manage site data: the earlier attempt gave me six)

then I took your link, and I'm not seeing anything

See [1767271](https://bugzilla.mozilla.org/show_bug.cgi?id=1767271)
- if you remove your clear cookies and site data exception for facebook (meaning you will have to login each time), does this still happen (might get slightly tricky to debug if you don't need a dFPI exception to login, IDK cuz I never use FB)

I was thinking about this just this morning. We currently have `network.cookie.thirdparty.sessionOnly` = true, but in the next release I'm commenting that out - IDK if the site exception to keep conflicts with that, it's a footgun and will probably be removed upstream. I was thinking about how we we-work the user.js  in #1441 for 101, before the upstream migration in likely 102, and the UI changes which won't apply in 101 but would in 102, and those UI changes are messy'ish

The other thing is, one assumes you are using uBO - add some rules to block some of these on third party sites globally. Personally, I have uBO set to block ALL 3p by default, so I'm sweet for my 4 exceptions, of which only github would possible ever be used as third party (and only on SSO login pages)

Also see `https://github.com/privacyguides/privacyguides.org/discussions/941`

**Comment by Thorin-Oakenpants** [comment]
hmm ... so is this two edged? If you allow a dFPI exception, does it then refuse to sanitize them?

anyway, I'll try remember to add something to #1441 about being selective about dFPI and sanitizing exceptions, but I'm hoping mozilla fix that bug ASAP, so I don;t have to.

Personally I think it should be top priority, because rolling out TCP for everyone might benefit the majority, but the UI actually makes this bug/hole very easy to replicate, and like I said elsewhere, imagine the most common dFPI exceptions = the biggest trackers on the internet, so well .. it won't look pretty as a bug AFTER the rollout

**Comment by remyabel2** [comment]
https://support.mozilla.org/en-US/kb/third-party-trackers?as=u&utm_source=inproduct#w_managing-cross-site-cookies

> While cross-site cookies from trackers are blocked in Firefox by default, a site may signal to the browser that it needs to use them for important functionality. In this case, Firefox will allow a third-party website to use cross-site cookies the first five times (or up to 1% of the number of unique sites you visit in a session, whichever is larger) without prompting you. After that, Firefox will prompt you to block these cookies. Without your consent, Firefox blocks these cookies from that point because a site requesting access that many times may be a tracker.

> Third-parties will only be able to prompt you if you interact with the website you are on. For example, if you visit [dogs.com](http://dogs.com/) and select the payment field, Amazon Pay cross-site cookies may be allowed to facilitate that transaction. After that, Firefox will ask you if you want to keep allowing them.

**Comment by EdwardLangdon** [comment]
What if I need to allow, X site but not facebook ?

The same setup as the Author of this issue.

**Comment by Thorin-Oakenpants** [comment]
> Oh - toggling this fixes it, mentioned in the other thread.

it does not fix it, it removes the d part of dFPI, and fucks everyone's dFPI exceptions. The solution is the bugzilla

**Comment by privacyguy123** [comment]
> > Oh - toggling this fixes it, mentioned in the other thread.
> 
> it does not fix it, it removes the d part of dFPI, and fucks everyone's dFPI exceptions. The solution is the bugzilla

With all due respect, the bugzilla you linked is 1 comment that doesn't offer any solution? (https://bugzilla.mozilla.org/show_bug.cgi?id=1767271) Do you mean when Mozilla get around to addressing and fixing it?

**Comment by Thorin-Oakenpants** [comment]
with all due respect, it is a BUG and fixing that BUG solves the issue - that's what bug fixing means

**Comment by Thorin-Oakenpants** [comment]
> How do you guys currently recommend to proceed with this problem

- resolve the bugzilla
- be judicious in what you keep - e.g. I only have six of which only one is used on other domains for SSO (github) - none are content providers like youtube - so if I really wanted to, I could use uBO to globally block those six domains except first party
- use containers/TC I guess, I have never bothered with them
- use a secondary browser for entertainment

FPI is not supported except for Tor Browser - example, we are not in PB mode and we have service workers which FPI doesn't cover. And if you were using FPI, CAD is still pointless

**Comment by s-rog** [comment]
Until that bug is fixed perhaps disabling sanitize on shutdown (just the cookies?) is an appropriate workaround?

**Comment by s-rog** [comment]
> And you would need to remove keep cookie + site data exceptions because the site exception is the issue

Ah yes forgot to mention that part.

> Depends on your view of 1st party.

I'm more concerned about cross party myself so I see this as a compromise and not a solution, hopefully the bug can be fixed soon (even though it's labeled as a feature on bugzilla).

Thanks for verifying my understanding!

**Comment by Thorin-Oakenpants** [comment]
for those not paying attention, it [sounds like](https://bugzilla.mozilla.org/show_bug.cgi?id=1767271#c4) this is slated to be addressed

Still confuses me how in settings when you open ETP>Exceptions it is blank (for me), but when you open Cookie+SiteData>Exceptions is is populated (for me with 5 entries) . Maybe ETP>Exceptions is something slightly different?

**Comment by Thorin-Oakenpants** [comment]
Also, ETP>Exceptions is getting an `Add` ability - landed in Nightly

**Comment by fxbrit** [comment]
[after reading this documentation](https://support.mozilla.org/en-US/kb/enhanced-tracking-protection-firefox-desktop) I get the impression that ETP exceptions only control the "block trackers" aspect of ETP. I tried to get a better understanding by looking at the source code but in all honesty I couldn't wrap my head around the permissions' code further than below notes:
  - ETP exceptions fall under the `trackingprotection` type -> https://searchfox.org/mozilla-central/source/browser/components/preferences/dialogs/permissions.js#325-334
  - there's also a `cookie` type in the same file
  - `trackingprotection` is used in [ContentBlockingAllowList.cpp](https://searchfox.org/mozilla-central/source/toolkit/components/antitracking/ContentBlockingAllowList.cpp#127)

so:
- pretty sure ETP exceptions don't overlap with cookie exceptions: if you whitelist a website from ETP, its cookies are not kept on close; according to 1767271 cookie exceptions overlap with Total Cookie Protection instead (and we already knew this).
- TCP had overlap with ETP but now it kinda doesn't since it's enabled in all modes. additionally, in the code ETP strict only flipped the `cookieBehaviour` pref with no other interaction with isolation, so I don't think the two are tightly coupled which would explain why their permissions don't overlap.

**Comment by Thorin-Oakenpants** [comment]
OK, I I confused this with dFPI vs ETP ... which are different

**Comment by Thorin-Oakenpants** [comment]
https://github.com/arkenfox/user.js/blob/b117916207862d4785f6da32d48cbe4420372434/user.js#L1009-L1012

repeat .. enabling FPI disables all network partitioning and TCP

**Comment by Thorin-Oakenpants** [comment]
almost three years now .. needs SUPER JESUS ....

**Comment by Jee-Hex** [comment]
Will Jesus save us (on Desktop) in time before the impending monstrosity known as nova arrives? I am already barely tolerating the latest round of settings "redesign"...

**Comment by any1here** [comment]
> Will Jesus save us (on Desktop) in time before the impending monstrosity known as nova arrives? I am already barely tolerating the latest round of settings "redesign"...

`browser.settings-redesign.enabled` to `false` until it eventually gets removed

**Comment by Jee-Hex** [comment]
Landed (I haven't bother testing) but not in time for the next esr...

**Comment by Thorin-II** [comment]
> Landed (I haven't bother testing) but not in time for the next esr...

yeah, I saw that and was thinking the same thing - so I left a suggestion
